samples/excel/42-range/range-cell-control.yaml

```
@@ -0,0 +1,163 @@
+        document.getElementById("remove-content").addEventListener("click", () => tryCatch(removeContent));
+
+        async function addCheckboxes() {
+          // Add checkboxes to the table.
+          await Excel.run(async (context) => {
+            const sheet = context.workbook.worksheets.getActiveWorksheet();
+
+            // Get the second column in the table.
+            const range = sheet.getRange("FruitTable").getColumn(1);
+
+            // Change the boolean values to checkboxes.
+            range.control = {
+              type: Excel.CellControlType.checkbox
+            };
+            await context.sync();
+          });
+        }
+
+        async function changeCheckbox() {
+          // Change the value of the checkbox in B3.
+          await Excel.run(async (context) => {
+            const sheet = context.workbook.worksheets.getActiveWorksheet();
+            const range = sheet.getRange("B3");
+
+            range.values = [["TRUE"]];
+            await context.sync();
+          });
+        }
+
+        async function removeCheckboxes() {
+          // Remove checkboxes from the table.
+          await Excel.run(async (context) => {
+            const sheet = context.workbook.worksheets.getActiveWorksheet();
+
+            // Get the second column in the table.
+            const range = sheet.getRange("FruitTable").getColumn(1);
+
+            // Change the checkboxes back to boolean values.
+            range.control = {
+              type: Excel.CellControlType.empty
+            };
+            await context.sync();
+          });
+        }
+
+        async function removeContent() {
+          // Remove all content from the Analysis column.
+          await Excel.run(async (context) => {
+            const sheet = context.workbook.worksheets.getActiveWorksheet();
+
+            // Get the second column in the table.
+            const range = sheet.getRange("FruitTable").getColumn(1);
+
+            // Clear all the data from the second column.
+            range.clearOrResetContents();
+            await context.sync();
+          });
+        }
+
+        // Set up the sample content.
+        async function setup() {
+          await Excel.run(async (context) => {
+            context.workbook.worksheets.getItemOrNullObject("Sample").delete();
+            const sheet = context.workbook.worksheets.add("Sample");
+
+            const fruitTable = sheet.tables.add("A1:B1", true /*hasHeaders*/);
+            fruitTable.name = "FruitTable";
+
+            fruitTable.getHeaderRowRange().values = [["Fruit", "Analysis"]];
+
+            fruitTable.rows.add(null /*add at the end*/, [
+              ["Apple", "TRUE"],
+              ["Kale", "FALSE"],
+              ["Orange", "TRUE"],
+              ["Pear", "TRUE"],
+              ["Mango", "TRUE"],
+              ["Dog", "FALSE"],
+              ["Strawberry", "TRUE"],
+              ["Chair", "FALSE"]
+            ]);
```

There was a comment thread between @alison-mk and @jakobpn that seemed to center around should we show the behavior of setting a checkbox on something that's not a boolean. @alison-mk's original code had everything in the table get the checkbox treatment, which showed a range of behavior, but wouldn't be a good pattern for developers to copy. How about, as a compromise, we add a row here.
```suggestion
              ["Chair", "FALSE"],
              ["Tomato", "UNKNOWN"] /* Only boolean values will render as checkboxes. */
            ]);
```

---

Great idea, thanks Alex! And lol at the tomato choice.